internal/storage/v2/memory/tenant.go

```diff
@@ -13,26 +14,36 @@ import (
 	"github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore"
 )
 
+var errInvalidMaxTraces = errors.New("max traces must be greater than zero")
+
 // Tenant is an in-memory store of traces for a single tenant
 type Tenant struct {
 	sync.RWMutex
-	ids        []pcommon.TraceID // ring buffer used to evict oldest traces
-	traces     map[pcommon.TraceID]ptrace.Traces
-	services   map[string]struct{}
-	operations map[string]map[tracestore.Operation]struct{}
-	config     *v1.Configuration
-	evict      int // position in ids[] of the last evicted trace
+	ids         map[pcommon.TraceID]int
+	traces      []tracesAndId // ring buffer used to store traces
+	services    map[string]struct{}
+	operations  map[string]map[tracestore.Operation]struct{}
+	config      *v1.Configuration
+	lastEvicted int // position in traces[] of the last evicted trace
 }
 
-func newTenant(cfg *v1.Configuration) *Tenant {
-	return &Tenant{
-		ids:        make([]pcommon.TraceID, cfg.MaxTraces),
-		traces:     map[pcommon.TraceID]ptrace.Traces{},
-		services:   map[string]struct{}{},
-		operations: map[string]map[tracestore.Operation]struct{}{},
-		config:     cfg,
-		evict:      -1,
+type tracesAndId struct {
```

Comment:
```suggestion
type traceAndId struct {
```
There's only one trace